File: regular_expressions/pl/subst_02.pl
#!/usr/bin/perl

##################################
# multiple pl substitutions
# @author: vladimir kulyukin
##################################

use strict;
use warnings;

my $txt0 = 'I like to program in Python.';
my $txt1 = 'I like to program in Perl.';


my @all_substs = [ ['I', 'We'], ['program', 'code'],
		   ['P\w+n', 'C'], ['.e\w+l', 'C#'] ];

sub make_substs {
    my ($substs, $txt) = @_;
    print 'Before substitution: ' . "\t" . $txt . "\n";    
    $txt =~ s/@{$_}[0]/@{$_}[1]/ foreach(@{$substs});
    print 'After substitution: '  . "\t" . $txt, "\n";
    print "******\n";
}

make_substs(\@all_substs, $txt0);
make_substs(\@all_substs, $txt1);

File: regular_expressions/py/subst_02.py
#!/usr/bin/python

############################
# multiple substitutions
# @author: vladimir kulyukin
############################

import re

txt0 = 'I like to program in Python.'
txt1 = 'I like to program in Perl.'

all_substs = ((r'I', 'We'), (r'program', 'code'),
              (r'P\w+n', 'C'), (r'.e\w+l', 'C#'))

def make_substs(substs, txt):
    print "Before substitution:\t" + txt
    temp_txt = txt
    for pat, repl in substs:
        temp_txt = re.sub(pat, repl, temp_txt)
    print "After substitution:\t" + temp_txt
    print '******'

make_substs(all_substs, txt0)
make_substs(all_substs, txt1)
